tests/deployment/switch_config.py

```diff
@@ -67,29 +68,28 @@ def configure():
 
 
 @pytest.fixture
-def not_dell_or_nexus():
-    """open the site-layout file to see if we don't have a dell or cisco nexus
-    switch"""
+def is_brocade():
+    """open the site-layout file to see if we have a brocade switch"""
 
     with open('site-layout.json') as layout_data:
         layout = json.load(layout_data)
     switch_type = layout['switches'][0]['type']
-    return (switch_type != NEXUS and switch_type != DELL5500 and
-            switch_type != DELLN3000)
+    return (switch_type == BROCADE)
 
 
-@pytest.mark.skipif(not_dell_or_nexus(),
-                    reason="Skipping because not a dell or cisco switch")
+@pytest.mark.skipif(is_brocade(), reason="Skipping because brocade switch")
 class TestSwitchSavingToFlash(NetworkTest):
-    """ saves the running config to the flash memory. Test is only for the dell
-        switch"""
+    """ saves the running config to the flash memory and tests if it succeeded
+    by comparing the running and startup config files"""
 
     def get_config(self, config_type):
-        """returns the switch configuration file"""
+        """helper method to get the switch config file by calling
+        the switch's get_config method
```

Comment:
Nitpick: get_config is defined on session, not switch.

Reply:
updated